pkg/operator/controller/nodeobservability/configmap.go

```diff
@@ -41,30 +40,27 @@ func (r *NodeObservabilityReconciler) createConfigMap(ctx context.Context, nodeO
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      configMapName.Name,
 			Namespace: configMapName.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					Name:       nodeObs.Name,
-					Kind:       nodeObs.Kind,
-					UID:        nodeObs.UID,
-					APIVersion: nodeObs.APIVersion,
-				},
-			},
 		},
 		Data: map[string]string{
 			kbltCAMountedFile: kbltCACM.Data[kbltCAMountedFile],
 		},
 	}
 
-	if err := r.Get(ctx, configMapName, &corev1.ConfigMap{}); err == nil {
-		return true, nil
-	} else if !errors.IsNotFound(err) {
-		return false, fmt.Errorf("error getting target configMap %s, %w", configMapName, err)
+	if err := controllerutil.SetControllerReference(nodeObs, configMap, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for target configmap %q: %w", configMapName, err)
 	}
 
-	if err := r.Create(ctx, configMap); err != nil {
-		return false, fmt.Errorf("failed to create target configMap %s: %w", configMapName, err)
+	if err := r.Get(ctx, configMapName, &corev1.ConfigMap{}); err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("error getting target configmap %q, %w", configMapName, err)
+		} else {
+			// create configmap since it is not found
+			if err := r.Create(ctx, configMap); err != nil {
+				return fmt.Errorf("failed to create target configmap %q: %w", configMapName, err)
+			}
+		}
 	}
 
-	r.Log.Info("created ConfigMap", "ConfigMap.Namespace", configMapName.Namespace, "ConfigMap.Name", configMapName.Name)
-	return true, nil
+	r.Log.Info("created kubelet CA configmap", "configmap.namespace", configMapName.Namespace, "configmap.name", configMapName.Name)
```

Comment:
This log has to be moved upper, to where the creation happens. Because if the target configmap is already there - no creation happens.